src/compiler/Stage/Optimize.elm

```diff
@@ -12,34 +13,41 @@ optimize project =
     Boilerplate.optimizeProject optimizeExpr project
 
 
-optimizeExpr : Typed.Expr -> Typed.Expr
-optimizeExpr expr =
+optimizeExpr : Typed.LocatedExpr -> Typed.LocatedExpr
+optimizeExpr =
```

Comment:
Please let's keep the argument in the function definition, like it was. I know it can be eta-reduced, but that goes contrary to the readability / simplicity / understandability values. I've even written a blogpost about stuff related to that a few years ago :smile: http://martin.janiczek.cz/clanek/being-clever-antipattern-in-elm/